### PR TITLE
Standardize license headers

### PR DESCRIPTION
**Pull request by ctubbsii**
This PR contains 3 commits:

1. Add build tooling to automatically generate missing license headers
2. Regenerate all the license headers for the project
3. Manual one-time cleanup that wasn't able to be automated

I'm not sure I want to redo this for 1.9, since it's a bit tedious, but it may help with getting things consistent. I'm open to feedback on that.

The main goal of this change is to standardize our style for license headers, and to automate adding missing ones during development.

### CODE REVIEW

**Comment by keith-turner** [comment]
If this works well its going to be awesome.  Does this new plugin run before the rat plugin?  Is the rat plugin still needed?

File: pom.xml [hunk 1]
```diff
@@ -775,6 +775,38 @@
...
+              <exclude>**/LICENSE</exclude>
+              <exclude>**/NOTICE</exclude>
+              <exclude>**/target/**</exclude>
+              <exclude>contrib/javadoc11.patch</exclude>
+            </excludes>
+            <mapping>
+              <HelloWorldTemplate>SLASHSTAR_STYLE</HelloWorldTemplate>
+              <Makefile>SCRIPT_STYLE</Makefile>
+              <TestTemplate>SLASHSTAR_STYLE</TestTemplate>
+              <accumulo-cluster>SCRIPT_STYLE</accumulo-cluster>
```
Comment (keith-turner) [inline pom.xml]: Is this setting a mapping for a maven module?

File: test/src/main/java/org/apache/accumulo/test/ShellIT.java [hunk 2]
```diff
@@ -400,7 +400,7 @@ public void execFileTest() throws IOException {
     Shell.log.debug("Starting exec file test --------------------------");
     shell.config("--config-file", config.toString(), "-u", "root", "-p", getRootPassword(), "-zi",
         getCluster().getInstanceName(), "-zh", getCluster().getZooKeepers(), "-f",
-        "src/main/resources/shellit.txt");
+        "src/main/resources/shellit.shellit");
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ShellIT.java]: Why did you need to rename this?

**Comment by milleruntime** [comment]
Can you make the license template match our current license headers (at least the majority)?  Then you could avoid re-generating one for every file.  I have never not copied and pasted one so I imagine they can't be all that different.

**Comment by ctubbsii** [comment]
> If this works well its going to be awesome. Does this new plugin run before the rat plugin? Is the rat plugin still needed?

Currently, it runs after the rat plugin. This plugin runs at `process-resources`, while rat runs at `validate` (see https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html for reference on lifecycle phases).

We can re-order if necessary, but I'm not sure it matters much.

**Comment by ctubbsii** [comment]
> Can you make the license template match our current license headers (at least the majority)? Then you could avoid re-generating one for every file. I have never not copied and pasted one so I imagine they can't be all that different.

I thought about that, but we had them too inconsistent, so the change was going to be large either way. I decided to converge on the format that is presented on ASF's official documentation instead.

File: test/src/main/java/org/apache/accumulo/test/ShellIT.java (hunk 2 above)
Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/ShellIT.java]: Because the filename extension, `.txt` has a specific format for the license header, built-in to the plugin, and the choice was either rename this test file (which I did), or override the built-in rule for *ALL* `.txt` files, which I didn't think was a good idea.

File: pom.xml (hunk 1 above)
Comment (ctubbsii) [reply pom.xml]: These mappings apply to specific files that appear in some of our maven modules. I could move these mappings to module-specific pom.xml configuration for the plugin, but it won't change the behavior of the plugin with respect to these files.

**Comment by keith-turner** [comment]
> Currently, it runs after the rat plugin.

It seems like it would be nice if it ran before the rat plugin, that way it could fix up files before rat fails the build.

File: test/src/main/java/org/apache/accumulo/test/ShellIT.java (hunk 2 above)
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ShellIT.java]: Thanks for the explanation.

**Comment by ctubbsii** [comment]
> > Currently, it runs after the rat plugin.
> 
> It seems like it would be nice if it ran before the rat plugin, that way it could fix up files before rat fails the build.

That makes sense. I'll do that.

File: pom.xml (hunk 1 above)
Comment (keith-turner) [reply pom.xml]: My question didn't even make sense, however I still found your answer useful.  I was confusing Accumulo with Fluo, Fluo had a fluo-cluster maven module.  Accumulo has no such module.  

I see the following.

```
$ find . -name "*accumulo-cluster*"
./assemble/bin/accumulo-cluster
```

**Comment by keith-turner** [comment]
Would it make sense to disable this new plugin for travis builds?  That way rat can fail the travis build if files need to be fixed.

**Comment by keith-turner** [comment]
> I'm not sure I want to redo this for 1.9, since it's a bit tedious, but it may help with getting things consistent. I'm open to feedback on that.

Is there any benefit to making the same changes in 1.9. I am assuming that we will only have merge conflicts when someone modifies the license portion in 1.9.  I think that type of modification in 1.9 would be very rare.  Also if there is conflict, could you just delete the license portion and just let the plugin put it back?

**Comment by ctubbsii** [comment]
> > I'm not sure I want to redo this for 1.9, since it's a bit tedious, but it may help with getting things consistent. I'm open to feedback on that.
> 
> Is there any benefit to making the same changes in 1.9.

The main benefit is just having the automation to add the new licenses. I'm not worried about conflicts, because it'd probably only happen, as you say, when the header itself was changed, which shouldn't really happen. For that alone, I'm disinclined to pull this back into 1.9. However, the secondary benefit is that it helps keep our build/release tooling sync'd between branches, which makes it easier to ensure consistent release quality across our actively maintained versions.

**Comment by keith-turner** [comment]
> which makes it easier to ensure consistent release quality across our actively maintained versions.

For this particular case having the rat plugin in 1.9 achieves the quality goal.

**Comment by ctubbsii** [comment]
> Would it make sense to disable this new plugin for travis builds? That way rat can fail the travis build if files need to be fixed.

Since we currently don't have any other formatting validation checks during CI, I wouldn't want to add it in this PR. However, that could be done as a follow-on to create a CI profile that we use for CI that skips all formatting, and does only formatting validation for all our automated formatting.

**Comment by ctubbsii** [comment]
> > which makes it easier to ensure consistent release quality across our actively maintained versions.
> 
> For this particular case having the rat plugin in 1.9 achieves the quality goal.

Aside from the manual fixes that I did to deal with double-license header issues in the bloomfilter code, I agree. I can backport those fixes manually on their own, though, without all the other changes.
